### ISSUE TITLE
Youtube theater mode not working on browser opening

### ISSUE THREAD

**Issue by jincastillo992** [brave/brave-browser#29591]
## Description 

When I open the browser app, the videos in youtube don't start in theater mode. This has being happening for about a week

## Steps to Reproduce

   1. Open browser
   2. Go to youtube
   3. Open any video

## Actual result:

![image](https://user-images.githubusercontent.com/31566251/230799132-28fb5c08-9d80-4853-94f9-1e2ef3605c6d.png)

## Expected result:
![image](https://user-images.githubusercontent.com/31566251/230799182-9849ce99-f2f1-4731-bea1-8d715853acd6.png)

## Reproduces how often: 

Always

## Brave version (brave://version info)

Brave | 1.50.114 Chromium: 112.0.5615.49 (Build oficial) (64 bits)
Revisión | bd2a7bcb881c11e8cfe3078709382934e3916914-refs/branch-heads/5615@{#936}
Sistema operativo | Windows 11 Version 22H2 (Build 22621.1413)

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?  No
- Does the issue resolve itself when disabling Brave Rewards? No
- Is the issue reproducible on the latest version of Chrome? No

**Comment by rebron** [comment]
@jincastillo992 Is there a setting in Youtube that allows you to start/view youtube videos in theatre mode? What you see in actual result is the default view.

**Comment by nathanpete1** [comment]
Does the expected result occur on other web browsers? 
If I were to guess, the status of what viewing mode you are using is not saved in cookies, and resets to default when launching the webpage from a fresh browser session

**Comment by jincastillo992** [comment]
> @jincastillo992 Is there a setting in Youtube that allows you to start/view youtube videos in theatre mode? What you see in actual result is the default view.
I have not touched any youtube settings. Yes, in the actual result is the default view, but, for example, in Chrome if you have a video playing in theater mode and you close the browser and open it again, the videos should play in theater mode



> Does the expected result occur on other web browsers? If I were to guess, the status of what viewing mode you are using is not saved in cookies, and resets to default when launching the webpage from a fresh browser session
I have Chrome installed and this doesn't happen there. I haven't touched any Brave settings or anything, I even reinstalled Brave this weekend thinking that may solve the problem, but no success.

**Comment by Vanerio** [comment]
The exact same thing has been happening to me for about a week, as you said.

**Comment by ErebusAres** [comment]
same issue, no settings changed, i dont think youtube even has optional settings anyway.

**Comment by RixzZ** [comment]
Same here, but I noticed also other changes on other websites. It's like if Brave was clearing some data on exit that it didn't clear before.

**Comment by d4kir92** [comment]
same problem

**Comment by sol3uk** [comment]
I've noticed this too and it's bugging me. Happy to help investigate if required

**Comment by d4kir92** [comment]
found a workaround till its fixed!

go to youtube, hit F12, go to console and put this in
document.cookie = 'wide=1; expires='+new Date('3099').toUTCString()+'; path=/';

press enter

now the cookie expires in year 3099

after restart brave the theater mode works normally, so it must be something with the cookies

**Comment by sol3uk** [comment]
Well yeah I mean that sorts it but doesn't explain if youtube has changed their cookie expiry or if brave released a change to how it treats the end of a "session" in which case it should be affecting more than just youtube.
![image](https://user-images.githubusercontent.com/26093813/235690158-4553a5d9-b505-4fd5-bf1f-401603810dc0.png)

**Comment by d4kir92** [comment]
I also lose some cookies on other websites with Brave
(i need to reaccept cookies sometimes)

**Comment by SiphoxR** [comment]
Thanks for the workaround. And I can confirm I lost cookies on two websites a few days ago without clearing cookies, and one of them was youtube iirc (can't remember the other one though)

**Comment by d4kir92** [comment]
k somehow that cookie got deleted, wtf

**Comment by d4kir92** [comment]
somehow the cookie only is alive 1 week, any fixes or ideas when this get fixed?

i also get cookie banners on sites where i am logged in and already accepted

brave settings for cookies on leave is nothing checked and rest of settings is also on default

**Comment by d4kir92** [comment]
so cookies get deleted after 7 days, so i need to manually add the cookie each week, great xD

**Comment by d4kir92** [comment]
still having this problem...

**Comment by d4kir92** [comment]
still this cookie gets deleted each week and not created automatically, so i need to manually add the cookie to get it working

this is not happening in chrome/edge

**Comment by asos-benashley** [comment]
I'm currently on **[Version 1.52.126 Chromium: 114.0.5735.133 (Official Build) (64-bit)](https://brave.com/latest/)**
And it's working fine for me. Was fine for me since trying the "wide" workaround above and it's remained in the browser since. This was only a youtube issue for me though, it didn't effect other sites like you mentioned above @d4kir92. You might need to look at browser settings around wiping cookies or just try a clean install.

**Comment by sol3uk** [comment]
I'm currently on **[Version 1.52.126 Chromium: 114.0.5735.133 (Official Build) (64-bit)](https://brave.com/latest/)**
And it's working fine for me. Was fine for me since trying the "wide" workaround above and it's remained in the browser since. This was only a youtube issue for me though, it didn't effect other sites like you mentioned above @d4kir92. You might need to look at browser settings around wiping cookies or just try a clean install.

**Comment by d4kir92** [comment]
> I'm currently on **[Version 1.52.126 Chromium: 114.0.5735.133 (Official Build) (64-bit)](https://brave.com/latest/)** And it's working fine for me. Was fine for me since trying the "wide" workaround above and it's remained in the browser since. This was only a youtube issue for me though, it didn't effect other sites like you mentioned above @d4kir92. You might need to look at browser settings around wiping cookies or just try a clean install.

yes, only youtube has the problem, but still happens each week
other sites are fine now

**Comment by Brave-Matt** [comment]
Lots of +1s here:
https://community.brave.com/t/youtube-theatre-mode-keeps-resetting-to-default-mode/481749/30

**Comment by ShivanKaul** [comment]
Hi folks, I'm trying to understand this issue. Is the issue that:

1. Theater mode goes away if you restart the browser? Or,
2. Theater mode goes away randomly, even if you haven't restarted the browser?

If it's 1: Given that `wide` is a session cookie (which seems to be how Youtube checks if you want Theater mode), I don't see how it could _ever_ last across a browser restart. Desktop Chrome behaves the exact same way. I see similar comments on Reddit about Chrome: https://www.reddit.com/r/youtube/comments/1063040/youtube_wont_automatically_go_into_theatre_mode/

If it's 2: the 7-day cookie expiry shouldn't apply to session cookies. I just confirmed that this is true and am adding a test for it: https://github.com/brave/brave-core/pull/19171

Would love steps to repro the problem such that it happens in Brave and not Chrome.

**Comment by goodov** [comment]
@ShivanKaul https://github.com/brave/brave-browser/issues/28379
cc @pes10k

**Comment by d4kir92** [comment]
> Hi folks, I'm trying to understand this issue. Is the issue that:
> 
> 1. Theater mode goes away if you restart the browser? Or,
> 2. Theater mode goes away randomly, even if you haven't restarted the browser?
> 
> If it's 1: Given that `wide` is a session cookie (which seems to be how Youtube checks if you want Theater mode), I don't see how it could _ever_ last across a browser restart. Desktop Chrome behaves the exact same way. I see similar comments on Reddit about Chrome: https://www.reddit.com/r/youtube/comments/1063040/youtube_wont_automatically_go_into_theatre_mode/
> 
> If it's 2: Brave's 7-day cookie expiry shouldn't apply to session cookies. I just confirmed that this is true and am adding a test for it: [brave/brave-core#19171](https://github.com/brave/brave-core/pull/19171). AFAICT YouTube's theater mode should behave the same on Brave and Chrome (and Edge, and...).
> 
> Would love steps to repro the problem such that it happens in Brave and not Chrome.

1. yes
2. not randomly

if i normally use youtube, if i set theatermode on it makes a wide cookie and if i restart its gone, but not on chrome/edge/...
if i manually add the wide cookie it stays 7 days

the link you sent says that it only happen when i am logged in, i am only logged in in brave

**Comment by Volodesi** [comment]
Same issue. Every time Brave is restarted. It forgets to stay in Theater mode. I don't clear or reset anything when closing/opening Brave.

Edge and Firefox on the same system having no issues with Theater mode staying between restarts.

**Comment by ShivanKaul** [comment]
Quick update here: the issue is that Chrome DOES persist session cookies across browser restarts if you have `Continue where you left off` setting on, which is not how session cookies are supposed to behave. We're looking into it.

> if i manually add the wide cookie it stays 7 days
How are you manually adding it?

**Comment by d4kir92** [comment]
go to youtube, hit F12, go to console and put this in
document.cookie = 'wide=1; expires='+new Date('3099').toUTCString()+'; path=/';

**Comment by ShivanKaul** [comment]
>document.cookie = 'wide=1; expires='+new Date('3099').toUTCString()+'; path=/';

Right, this will last only 7 days because you're specifying the `expires` and we cap the maximum expires on cookies set via JS.

**Comment by d4kir92** [comment]
So we can make a Cookie without expires?

**Comment by ShivanKaul** [comment]
We're introducing a fix for YouTube to let the `wide` session cookie persist beyond a browser restart.

**Comment by d4kir92** [comment]
it still happening

**Comment by ShivanKaul** [comment]
@d4kir92 could you give a video recording of the issue?

**Comment by d4kir92** [comment]
Brave Ad Block Updater - Version: 1.0.372

**Comment by d4kir92** [comment]
https://www.youtube.com/watch?v=rb8vO6uCUN0

**Comment by d4kir92** [comment]
well yes, it get deleted everytime i close the browser

**Comment by ShivanKaul** [comment]
If you manually added a cookie like you said you did in https://github.com/brave/brave-browser/issues/29591#issuecomment-1624189997, that would not get deleted when you close the browser.

**Comment by d4kir92** [comment]
i didnt added the cookie the last 10 days or so, i also removed the one that get created by youtube and tested, same result

**Comment by ShivanKaul** [comment]
Just to follow-up here, we discussed this at https://community.brave.com/t/problem-youtube-cookie-gets-deleted-each-week-also-other-sites/489465/30. Please be sure to have Brave Shields ON, otherwise our fix won't work.

**Comment by Emi-TheDhamphirInLoveUnderTheFrozenStar** [comment]
@ShivanKaul
I tested this and it is working, the problem is Youtube, and that's probably what people are experiencing. I know this because I set a cookie `PREF gl=US` with Brave's adblocker and that location only takes action **after** Youtube gets reloaded.
That's bad but not too bad, it is a behavior intended by Youtube, because they even do it, for example, if Dark theme gets detected by youtube, Youtube will refresh itself with the url: `youtube.com/?themeRefresh=1` to fit the Browser's dark or light them with its `Use Device theme`.
So in my cause, if I go to youtube In Private window, I can see the USA location right away, because youtube is refreshing itself because of the detection of theme. That means, if I were to set the whole thing `f6=40000000&gl=US` then I will have to refresh Youtube myself for everything to take action, because Youtube is not going to do it then.

Youtube knows it needs to restart itself to take these cookies into consideration, and they know it if they are using it as a way to apply dark or light theme.

So saying this, I am 100% sure `brave-youtube-theater-fix`.
But as explained, if the cookie gets remove when youtube is closed, which means the localstorage stays as value of 1 or 0, and the script will set the cookie `wide=1` when a user opens youtube, Youtube will only use the cookie until a refresh happens, and that's what people will not see even with this workaround, and why I say that might be the reason why people say it is not working.

Of course, you can only see this only in Devtools and understating how Youtube works, which means users would have to set cookies and localstorage items like I do, so I don't blame them.
This means that being a Youtube behavior, can there be a fix? the only solution would be to cause a refresh in the page when the theater cookie doesn't exist and it is set by the script, but then, it should be the users refreshing the page.

Basically the steps to see this easy is to:
1. go to any youtube video and hit t
2. close youtube
3. go to `brave://settings/cookies/detail?site=youtube.com` and delete wide
4. open youtube.com go to any video and you will see it doesn't start in theater mode.
5. refresh and it will be in theater mode.

**Comment by d4kir92** [comment]
still happening!

**Comment by MadLad1Clip** [comment]
Go to chrome settings > Privacy & Security > Thrid Party Cookies and add YouTube to the allowed to use cookies section.

**Comment by Rokeugon** [comment]
ive tried with brave shield on and it still doesn't work. ive tried manually editing the cookies session 

![Image](https://github.com/user-attachments/assets/4804026b-1c34-4a00-b128-71a0369880c0)

ive tried manually typing out the long format and just putting in the year itself. it just always auto corrects itself. it works for a couple days then resets itself and im stuck on default mode again.

how do you make this permanent im sick of applying this fix every 3 days
